### PR TITLE
add login.gov auth and create harvest user table

### PR DESCRIPTION
**Pull request by Jin-Sun-tts**
# Pull Request

Related to https://github.com/GSA/data.gov/issues/4674

## About

- Created a harvest_user table to store registered users.
- Created new apps in the login.gov sandbox for development, staging, and production environments.
- Implemented authentication using login.gov to secure the add, edit, and delete routes for organizations and harvest sources.
- Developed Flask CLI commands to manage users, including adding, listing, and removing users.
- Added secrets to the cloud service, including the Flask app secret key and the private key for OpenID.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -54,3 +54,5 @@ websocket-client==1.7.0 ; python_version >= "3.10"
 werkzeug==3.0.2 ; python_version >= "3.10"
 wtforms==3.1.2 ; python_version >= "3.10"
 yarl==1.9.4 ; python_version >= "3.10"
+PyJWT
```
Comment (btylerburton) [inline requirements.txt:57]: you can run `make deps-update` and it should add the version number in correct format

File: .env [hunk 2]
```diff
@@ -10,3 +10,9 @@ HARVEST_SOURCE_URL=http://localhost:${HARVEST_SOURCE_PORT}
 
 CF_API_URL=https://api.fr.cloud.gov
 LM_RUNNER_APP_GUID=f4ab7f86-bee0-44fd-8806-1dca7f8e215a
+
+CLIENT_ID=urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:datagov-dev-harvest-admin
+ISSUER=https://idp.int.identitysandbox.gov
+REDIRECT_URI=http://localhost:8080/callback
+FLASK_APP_SECRET_KEY=Px4o0MPEsCzHJRRSUDyt1VRi-lYF2W5VEzR13lylAgI
```
Comment (btylerburton) [inline .env:17]: will need to recreate these keys in dev I'm guessing

File: vars.prod.yml [hunk 3]
```diff
@@ -1,4 +1,8 @@
 app_name: harvesting-logic
 database_name: harvesting-logic-db
 
-route-external: harvester-prod-datagov.app.cloud.gov
\ No newline at end of file
+route-external: harvester-prod-datagov.app.cloud.gov
+
+CLIENT_ID: urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:datagov-prod-harvest-admin
+ISSUER: https://idp.int.identitysandbox.gov
```
Comment (btylerburton) [inline vars.prod.yml:7]: will need prod and staging values

**Review by btylerburton — DISMISSED** [review]
Nice work! Can't wait to get this integrated.

File: vars.prod.yml (hunk 3 above)
Comment (Jin-Sun-tts) [reply vars.prod.yml:7]: right, we will need make request to have it in prod and staging, for now we just host it in the sandbox.

File: .env (hunk 2 above)
Comment (Jin-Sun-tts) [reply .env:17]: right, for now we just keep it same in local and dev. We will create new one on staging and prod later.

File: requirements.txt (hunk 1 above)
Comment (Jin-Sun-tts) [reply requirements.txt:57]: done. Thanks

**Review by btylerburton — APPROVED** [review]

**Pull request merged**